src/Octoshift/GithubApi.cs

```diff
@@ -304,5 +304,34 @@ public virtual async Task<bool> RevokeMigratorRole(string org, string actor, str
                 return false;
             }
         }
+
+        public virtual async Task<int> StartArchiveGeneration(string apiUrl, string org, string[] repositories)
+        {
+            var url = $"{apiUrl}/orgs/{org}/migrations";
+
+            var payload = new { repositories };
+
+            var response = await _client.PostAsync(url, payload);
+            var data = JObject.Parse(response);
+            return (int)data["id"];
+        }
+
+        public virtual async Task<string> MigrationStatus(string apiUrl, string org, int migrationId)
```

Comment:
Two things here:
1.  By convention method names in `C#` usually have the verb in them to show that they are performing some action. So here a better name would start with `Get` i.e. `GetMigrationStatus`
2. We already have a `GetMigrationState` method which may easily be confused with this one. I am guessing this is returning the migration status for the archive generation? if so then it's better to name it accordingly, i.e. `GetArchiveGenerationStatus`.

Reply:
Updated the name, and we bass in the apiUrl because for ghes and ghae we'll need it since it's not the default 👍